src/crypto/index.ts

```diff
@@ -2866,19 +2866,28 @@ export class Crypto extends TypedEventEmitter<CryptoEvent, CryptoEventHandlerMap
      */
     public async decryptEvent(event: MatrixEvent): Promise<IEventDecryptionResult> {
         if (event.isRedacted()) {
+            // Try to decrypt the redaction event, to support encrypted
+            // redaction reasons.  If we can't decrypt, just fall back to using
+            // the original redacted_because.
             const redactionEvent = new MatrixEvent({
                 room_id: event.getRoomId(),
                 ...event.getUnsigned().redacted_because,
             });
-            const decryptedEvent = await this.decryptEvent(redactionEvent);
+            let redactedBecause: IEvent;
+            try {
+                const decryptedEvent = await this.decryptEvent(redactionEvent);
+                redactedBecause = decryptedEvent.clearEvent as IEvent;
+            } catch {
+                redactedBecause = event.getUnsigned().redacted_because!;
+            }
```

Comment:
I feel like we should make this conditional on `redactionEvent.isEncrypted()`, and log a warning if we find it is encrypted but we can't decrypt.